battleship-client/src/test/java/com/epam/solid/nie/client/ui/ShipTest.java

```diff
@@ -0,0 +1,51 @@
+package com.epam.solid.nie.client.ui;
+
+import com.epam.solid.nie.ships.HorizontalShipFactory;
+import com.epam.solid.nie.ships.ShipFactory;
+import com.epam.solid.nie.utils.Point2D;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Test
+public class ShipTest {
+
+    public void should_removeHealth_whenHit(){
+        //given
+        ShipFactory shipFactory = new HorizontalShipFactory();
+        ShipCreator shipCreator = new ShipCreator(shipFactory);
+        List<Cell> cells = Arrays.asList(new Cell(Point2D.of(0, 0)), new Cell(Point2D.of(0, 1)), new Cell(Point2D.of(3, 0)), new Cell(Point2D.of(1, 1)));
+        Ship ship = shipCreator.createShip(cells);
+
+        //when
+        ship.hit(Point2D.of(0,0));
+        ship.hit(Point2D.of(0,1));
+        ship.hit(Point2D.of(3,0));
+        ship.hit(Point2D.of(1,1));
+        ship.hit(Point2D.of(1,1));
+
+        //then
+        Assert.assertEquals(ship.getRemainingHealth(), 0);
+    }
+
+    public void should_TellIfShipIsAlive(){
```

Comment:
shouldn't there be another test of this kind to test whether method return true in appropriate case